src/Integration.Vsix.UnitTests/CFamily/CmdBuilderTests.cs

```diff
@@ -316,21 +232,29 @@ public void AddOptFromProperties(string input, string output, string cmd)
             cmdBuilder.GetFullCmd().Should().Be(cmd);
         }
 
-        [TestMethod]
-        [DataRow("Default", "")]
-        [DataRow("CompileAsC", "/TC ")]
-        [DataRow("CompileAsCpp", "/TP ")]
-        public void HeaderFileLang(string compileAs, string cmd)
+        [DataTestMethod]
+        [DataRow(false, "", "", "")]
+        [DataRow(true, "", "", "")]
+        [DataRow(false, "1", "", "1 ")]
+        [DataRow(true, "1", "", "1 ")]
+        [DataRow(false, "", "2", "2 ")]
```

Comment:
Minor comment:

The real `LanguageFlagsProvider` seems to always return a non-empty string for the compile-as flag (i.e. either `/TC` or `/TP`, but never blank), as described in the PR summary. This implies that the expected command should always have either of these.

I was confused by the expected "2 " here (which means that no `/TC` or `/TP` were added). As far as I understand, this mock in this test doesn't behave similar to the real implementation.

Would it make sense to somehow align the behavior of the mock here with the real implementation by always returning a non-empty for `compileAsFlag`?

(I would leave it for you to decide if it is worth it, I am just sharing it because it took me a bit to understand what is going on)

Reply:
The idea of the test was more to show that the implementation for choosing the values is delegated and should be tested in the LanguageFlagsProviderTests. I will just change the values to be more readable than just 1 & 2.